client/document/index.jsx

```diff
@@ -110,15 +110,19 @@ class Document extends Component {
 		let headTitle = head.title;
 		let headFaviconUrl;
 
-		// To customize the page title and favicon for the Gravatar passwordless login relevant pages.
+		// To customize the page title and favicon for Gravatar-related login pages.
 		if ( sectionName === 'login' && typeof query?.redirect_to === 'string' ) {
 			const searchParams = new URLSearchParams( query.redirect_to.split( '?' )[ 1 ] );
-			// Get the client ID from the redirect URL to cover the case of a login URL without the "client_id" parameter, e.g. /log-in/link/use
-			const oauth2Client = initialClientsData[ searchParams.get( 'client_id' ) ] || {};
-			const isGravPoweredClient = isGravPoweredOAuth2Client( oauth2Client );
+			// Use Gravatar's favicon + title for the Gravatar-related OAuth2 clients in SSR.
+			const oauth2Client = query?.gravatar_flow
+				? gravatarClientData
+				: // To cover the case where the `client_id` is not provided, e.g. /log-in/link/use
+				  initialClientsData[ searchParams.get( 'client_id' ) ] || {};
 
-			headTitle = isGravPoweredClient ? oauth2Client.title : headTitle;
-			headFaviconUrl = isGravPoweredClient ? oauth2Client.favicon : headFaviconUrl;
+			if ( isGravPoweredOAuth2Client( oauth2Client ) ) {
+				headTitle = oauth2Client.title;
+				headFaviconUrl = oauth2Client.favicon;
+			}
```

Comment:
I would replace lines 116 to 125 with this (I can't suggest a change because of deleted lines)

```
			if ( isGravPoweredOAuth2Client( oauth2Client ) ) {
				headTitle = gravatarClientData.title;
				headFaviconUrl = gravatarClientData.favicon;
			}
```

Reply:
✅ It's addressed through [280b39e](https://github.com/Automattic/wp-calypso/pull/93133/commits/280b39e013ac391a49e6b6305c8754377e7926dd)